torchrl/modules/tensordict_module/sequence.py

```diff
@@ -235,273 +143,11 @@ def select_subsequence(
-        **kwargs,
-    ) -> TensorDictBase:
-        if params is not None and buffers is not None:
-            if isinstance(params, TensorDictBase):
-                # TODO: implement sorted values and items
-                param_splits = list(zip(*sorted(params.items())))[1]
-                buffer_splits = list(zip(*sorted(buffers.items())))[1]
-            else:
-                param_splits = self._split_param(params, "params")
-                buffer_splits = self._split_param(buffers, "buffers")
-            for i, (module, param, buffer) in enumerate(
-                zip(self.module, param_splits, buffer_splits)
-            ):
-                if "vmap" in kwargs and i > 0:
-                    # the tensordict is already expended
-                    if not isinstance(kwargs["vmap"], tuple):
-                        kwargs["vmap"] = (0, 0, *(0,) * len(module.in_keys))
-                    else:
-                        kwargs["vmap"] = (
-                            *kwargs["vmap"][:2],
-                            *(0,) * len(module.in_keys),
-                        )
-                tensordict = self._run_module(
-                    module, tensordict, params=param, buffers=buffer, **kwargs
-                )
-
-        elif params is not None:
-            if isinstance(params, TensorDictBase):
-                # TODO: implement sorted values and items
-                param_splits = list(zip(*sorted(params.items())))[1]
-            else:
-                param_splits = self._split_param(params, "params")
-            for i, (module, param) in enumerate(zip(self.module, param_splits)):
-                if "vmap" in kwargs and i > 0:
-                    # the tensordict is already expended
-                    if not isinstance(kwargs["vmap"], tuple):
-                        kwargs["vmap"] = (0, *(0,) * len(module.in_keys))
-                    else:
-                        kwargs["vmap"] = (
-                            *kwargs["vmap"][:1],
-                            *(0,) * len(module.in_keys),
-                        )
-                tensordict = self._run_module(
-                    module, tensordict, params=param, **kwargs
-                )
-
-        elif not len(kwargs):
-            for module in self.module:
-                tensordict = self._run_module(module, tensordict, **kwargs)
-        else:
-            raise RuntimeError(
-                "TensorDictSequential does not support keyword arguments other than 'tensordict_out', 'in_keys', 'out_keys' 'params', 'buffers' and 'vmap'"
-            )
-        if tensordict_out is not None:
-            tensordict_out.update(tensordict, inplace=True)
-            return tensordict_out
-        return tensordict
-
-    def __len__(self):
-        return len(self.module)
+        td_sequential = super().select_subsequence(in_keys=in_keys, out_keys=out_keys)
+        return TensorDictSequential(*td_sequential.module)
```

Comment:
I guess the only difference is that it returns an object of the same type as the parent?
Maybe we could modify the original `select_subsequence` in tensordict?

Reply:
Yes you're right.

I think if we change `tensordict.nn.TensorDictSequential.select_subsequence` to return `self.__class__(...)` then we can get rid of this method here.